pymeasure/instruments/attocube/anc300.py

```diff
@@ -255,3 +246,109 @@ def stop_all(self):
             attribute = getattr(self, attr)
             if isinstance(attribute, Axis):
                 attribute.stop()
+
+    def __init__(
+        self,
+        adapter=None,
+        name="attocube ANC300 Piezo Controller",
+        axisnames="",
+        passwd="",
+        query_delay=0.05,
+        **kwargs,
+    ):
+        """
+        The function is called with the following arguments:
+
+        The function checks if the adapter is None, and if so, it checks if the host is defined.
+        If the host is defined, it creates a VISA socket connection.
+
+        .. deprecated:: 0.11.2
+            The 'host' argument is deprecated. Use 'adapter' instead.
+
+        :param adapter: The address of the controller
+        :param name: The name of the device, defaults to attocube ANC300 Piezo Controller (optional)
+        :param axisnames: a string of the axis names, e.g. "xyz"
+        :param passwd: the password for the controller
+        :param query_delay: the time to wait between queries to the controller
+        """
+        if adapter is None:
+            adapter = self.handle_deprecated_host_arg(kwargs)
+
+        self.query_delay = query_delay
+        self.termination_str = "\r\n"
+
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=False,
+            read_termination=self.termination_str,
+            write_termination=self.termination_str,
+            **kwargs
+        )
+
+        self._axisnames = axisnames
+        for i, axis in enumerate(axisnames):
+            setattr(self, axis, Axis(self, i + 1))
+
+        self.wait_for()
+        super().read()  # clear messages sent upon opening the connection
+        # send password and check authorization
+        self.write(passwd)
+        self.wait_for()
+        ret: str = super().read()
+        auth_msg = ret.split(self.termination_str)[1]
+        if auth_msg != 'Authorization success':
+            raise Exception(f"Attocube authorization failed '{auth_msg}'")
+        # switch console echo off
+        self.write('echo off')
+        _ = self.read()
+
+    def handle_deprecated_host_arg(self, kwargs: dict):
+        """
+        This pulls the host argument from the kwargs and creates a resource string
+        for the VISAAdapter. This is deprecated and separated out to make it easier
+        to remove in the future. This function should be removed and the adapter
+        argument should be made non-optional in the definition of the __init__ function.
+
+        .. deprecated:: 0.11.2
+            The 'host' argument is deprecated.
+        """
```

Comment:
We do not use type hinting in pymeasure. You can put that information in the docstring:
":param dict \\**kwargs: Keyword arguments..."

I'd put the deprecated host argument in the class docstring, as that way it is visible to everyone creating an instance of that class.

Reply:
Thanks, done